diff --git a/accel/tcg/translate-all.c b/accel/tcg/translate-all.c
index d468667b0d..fb7a0b69f4 100644
--- a/accel/tcg/translate-all.c
+++ b/accel/tcg/translate-all.c
@@ -273,10 +273,10 @@ TranslationBlock *tb_gen_code(CPUState *cpu, TCGTBCPUState s)
 
     phys_pc = get_page_addr_code_hostp(env, s.pc, &host_pc);
 
-    if (phys_pc == -1) {
+    //if (phys_pc == -1) {
         /* Generate a one-shot TB with 1 insn in it */
         s.cflags = (s.cflags & ~CF_COUNT_MASK) | 1;
-    }
+    //}
 
     max_insns = s.cflags & CF_COUNT_MASK;
     if (max_insns == 0) {
diff --git a/linux-user/elfload.c b/linux-user/elfload.c
index 26773e6c31..672fe36c0c 100644
--- a/linux-user/elfload.c
+++ b/linux-user/elfload.c
@@ -168,12 +168,14 @@ typedef abi_int         target_pid_t;
 
 #define ELF_HWCAP get_elf_hwcap()
 
+#if 0
 static uint32_t get_elf_hwcap(void)
 {
     X86CPU *cpu = X86_CPU(thread_cpu);
 
     return cpu->env.features[FEAT_1_EDX];
 }
+#endif
 
 #ifdef TARGET_X86_64
 #define ELF_CLASS      ELFCLASS64
@@ -2448,7 +2450,7 @@ static abi_ulong setup_arg_pages(struct linux_binprm *bprm,
     if (info->exec_stack) {
         prot |= PROT_EXEC;
     }
-    error = target_mmap(0, size + guard, prot,
+    error = target_mmap(0x3ff69fc000, size + guard, prot,
                         MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
     if (error == -1) {
         perror("mmap stack");
@@ -2581,8 +2583,8 @@ static abi_ulong create_elf_tables(abi_ulong p, int argc, int envc,
     int i;
     abi_ulong u_rand_bytes;
     uint8_t k_rand_bytes[16];
-    abi_ulong u_platform, u_base_platform;
-    const char *k_platform, *k_base_platform;
+    abi_ulong u_platform/*, u_base_platform*/;
+    const char *k_platform/*, *k_base_platform*/;
     const int n = sizeof(elf_addr_t);
 
     sp = p;
@@ -2604,6 +2606,7 @@ static abi_ulong create_elf_tables(abi_ulong p, int argc, int envc,
         }
     }
 
+#if 0
     u_base_platform = 0;
     k_base_platform = ELF_BASE_PLATFORM;
     if (k_base_platform) {
@@ -2619,6 +2622,7 @@ static abi_ulong create_elf_tables(abi_ulong p, int argc, int envc,
             sp += len + 1;
         }
     }
+#endif
 
     u_platform = 0;
     k_platform = ELF_PLATFORM;
@@ -2648,7 +2652,15 @@ static abi_ulong create_elf_tables(abi_ulong p, int argc, int envc,
     /*
      * Generate 16 random bytes for userspace PRNG seeding.
      */
+#if 1
+    int *ptr = (int *)k_rand_bytes;
+    ptr[0] = 0x55aa;
+    ptr[1] = 0x55aa;
+    ptr[2] = 0x55aa;
+    ptr[3] = 0x55aa;
+#else
     qemu_guest_getrandom_nofail(k_rand_bytes, sizeof(k_rand_bytes));
+#endif
     if (STACK_GROWS_DOWN) {
         sp -= 16;
         u_rand_bytes = sp;
@@ -2660,10 +2672,16 @@ static abi_ulong create_elf_tables(abi_ulong p, int argc, int envc,
         sp += 16;
     }
 
+#if 0
     size = (DLINFO_ITEMS + 1) * 2;
+#else
+    size = (20 + 1) * 2;
+#endif
+#if 0
     if (k_base_platform) {
         size += 2;
     }
+#endif
     if (k_platform) {
         size += 2;
     }
@@ -2673,8 +2691,10 @@ static abi_ulong create_elf_tables(abi_ulong p, int argc, int envc,
 #ifdef DLINFO_ARCH_ITEMS
     size += DLINFO_ARCH_ITEMS * 2;
 #endif
+#if 0
 #ifdef ELF_HWCAP2
     size += 2;
+#endif
 #endif
     info->auxv_len = size * n;
 
@@ -2718,6 +2738,34 @@ static abi_ulong create_elf_tables(abi_ulong p, int argc, int envc,
     /* There must be exactly DLINFO_ITEMS entries here, or the assert
      * on info->auxv_len will trigger.
      */
+#if 1
+    if (vdso_info) {
+        NEW_AUX_ENT(AT_SYSINFO_EHDR, vdso_info->load_addr);
+    }
+    NEW_AUX_ENT(AT_MINSIGSTKSZ, 0x5a0);
+    NEW_AUX_ENT(AT_HWCAP, (abi_ulong)0x00000000178bfbfdUL);
+    NEW_AUX_ENT(AT_PAGESZ, (abi_ulong)(TARGET_PAGE_SIZE));
+    NEW_AUX_ENT(AT_CLKTCK, (abi_ulong)0x64);
+    NEW_AUX_ENT(AT_PHDR, (abi_ulong)(info->load_addr + exec->e_phoff));
+    NEW_AUX_ENT(AT_PHENT, (abi_ulong)(sizeof (struct elf_phdr)));
+    NEW_AUX_ENT(AT_PHNUM, (abi_ulong)(exec->e_phnum));
+    NEW_AUX_ENT(AT_BASE, (abi_ulong)0);
+    NEW_AUX_ENT(AT_FLAGS, (abi_ulong)0);
+    NEW_AUX_ENT(AT_ENTRY, info->entry);
+    NEW_AUX_ENT(AT_UID, (abi_ulong) getuid());
+    NEW_AUX_ENT(AT_EUID, (abi_ulong) geteuid());
+    NEW_AUX_ENT(AT_GID, (abi_ulong) getgid());
+    NEW_AUX_ENT(AT_EGID, (abi_ulong) getegid());
+    NEW_AUX_ENT(AT_SECURE, (abi_ulong)0);
+    NEW_AUX_ENT(AT_RANDOM, (abi_ulong) u_rand_bytes);
+    NEW_AUX_ENT(AT_HWCAP2, (abi_ulong)0);
+    NEW_AUX_ENT(AT_EXECFN, info->file_string);
+    if (u_platform) {
+        NEW_AUX_ENT(AT_PLATFORM, u_platform);
+    }
+    NEW_AUX_ENT(AT_RSEQ_FEATURE_SIZE, 28);
+    NEW_AUX_ENT(AT_RSEQ_ALIGN, 32);
+#else
     NEW_AUX_ENT(AT_PHDR, (abi_ulong)(info->load_addr + exec->e_phoff));
     NEW_AUX_ENT(AT_PHENT, (abi_ulong)(sizeof (struct elf_phdr)));
     NEW_AUX_ENT(AT_PHNUM, (abi_ulong)(exec->e_phnum));
@@ -2752,6 +2800,7 @@ static abi_ulong create_elf_tables(abi_ulong p, int argc, int envc,
     if (vdso_info) {
         NEW_AUX_ENT(AT_SYSINFO_EHDR, vdso_info->load_addr);
     }
+#endif
     NEW_AUX_ENT (AT_NULL, 0);
 #undef NEW_AUX_ENT
 
